config/helpers/generator.py

```diff
@@ -1,15 +1,62 @@
-from config.datamodel import LlamaFarmConfig, Version
+from pathlib import Path
+from typing import Any
 
+try:
+    import yaml  # type: ignore
+except ImportError:  # pragma: no cover
+    yaml = None
 
-def generate_base_config():
-    return LlamaFarmConfig(
-        version=Version.v1,
-        name="llamafarm",
-        rag={},
-        prompts=[],
-        datasets=[],
-        models=[],
-    ).model_dump(mode="json")
+from config.datamodel import LlamaFarmConfig
 
-# Example usage:
-# generate_base_config_from_schema("schema.yaml", "llamafarm.yaml")
+
+def generate_base_config() -> dict:
+    """
+    Deprecated shim that now proxies to the schema-based generator using
+    the built-in `config/schemas/default.yaml`. The returned config name
+    is overridden to `llamafarm` for backward compatibility with callers
+    that set project name after generation.
+    """
+    default_schema = (
+        Path(__file__).parent.parent / "schemas" / "default.yaml"
+    )
+    cfg = generate_base_config_from_schema(str(default_schema))
+    cfg.update({"name": "llamafarm"})
+    return cfg
+
+
+def generate_base_config_from_schema(schema_path: str) -> dict:
```

Comment:
There is no need to have two separate functions here. Just add an optional extra param to the original `generate_base_config` function and use that throughout the codebase